packages/sanity/src/core/form/members/object/fields/ArrayOfObjectsField.tsx

```diff
@@ -99,7 +102,7 @@ export function ArrayOfObjectsField(props: {
 
       if (isRemovingLastItem) {
         // apply the patch to the current value
-        const result = applyAll(member.field.value || [], patches)
+        const result = applyAll(valueRef.current || [], patches)
```

Comment:
I'm gonna update the value ref here also as according to what was done by @RitaDias here: https://github.com/sanity-io/sanity/pull/4518/files#diff-bb9bedb1a67a3a5e89f968e104a98a4df4770f736cb430c59d53a93af8170242R46-R49